src/binary/writer.rs

```diff
@@ -868,6 +860,76 @@ impl<W: Write> BinarySystemWriter<W> {
     }
 }
 
+/// These functions provide "raw" encoding steps of values into Ion binary.
+///
+/// For example, consider an i64 as defined in [the spec][spec]:
+///
+/// ```text
+///            7       4 3       0
+///           +---------+---------+
+/// Int value |  2 or 3 |    L    |
+///           +---------+---------+======+
+///           :     length [VarUInt]     :
+///           +==========================+
+///           :     magnitude [UInt]     :
+///           +==========================+
+/// ```
+///
+/// There are 3 steps here:
+///
+/// 1. The type (positive or negative int)
+/// 2. The length of the binary representation of the int
+/// 3. The magnitude of the int
+///
+/// To represent an i64 in Ion binary, all 3 steps need to be taken. However, to
+/// include an i64 in an Ion hash computation, the length is not relevant!
+/// That's why this factoring is useful: it enables code sharing of the
+/// individual steps.
+///
+/// [spec]: https://amzn.github.io/ion-docs/docs/binary.html#2-and-3-int
+pub mod raw {
+    use super::*;
+
+    /// The result of `i64_magnitude`. The fields are private; users must go
+    /// through `as_ref()` to see the slice that ignores leading zeros.
+    ///
+    /// This type exists to avoid heap allocation.
+    pub struct I64Magnitude {
```

Comment:
Should this `#[derive(Copy, Clone, Debug)]` at a minimum?

Reply:
I think not - I literally want to hide anything other than the slice view.

Reply:
I landed up adding this, but I'm not sure it has a use.